packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md

```diff
@@ -0,0 +1,816 @@
+}
+```
+
+</details>
+<details>
+<summary>Diff patch from v7 to v8 for <em><code>stylistic-type-checked</code></em></summary>
+
+```diff
+{
+   '@typescript-eslint/adjacent-overload-signatures': '...',
+   '@typescript-eslint/array-type': '...',
+   '@typescript-eslint/ban-tslint-comment': '...',
+   '@typescript-eslint/class-literal-property-style': '...',
+   '@typescript-eslint/consistent-generic-constructors': '...',
+   '@typescript-eslint/consistent-indexed-object-style': '...',
+   '@typescript-eslint/consistent-type-assertions': '...',
+   '@typescript-eslint/consistent-type-definitions': '...',
+   'dot-notation': '...',
+   '@typescript-eslint/dot-notation': '...',
+   '@typescript-eslint/no-confusing-non-null-assertion': '...',
+   'no-empty-function': '...',
+   '@typescript-eslint/no-empty-function': '...',
+-  '@typescript-eslint/no-empty-interface': '...',
+   '@typescript-eslint/no-inferrable-types': '...',
+   '@typescript-eslint/non-nullable-type-assertion-style': '...',
++  '@typescript-eslint/prefer-find': '...',
+   '@typescript-eslint/prefer-for-of': '...',
+   '@typescript-eslint/prefer-function-type': '...',
++  '@typescript-eslint/prefer-includes': '...',
+-  '@typescript-eslint/prefer-namespace-keyword': '...',
+   '@typescript-eslint/prefer-nullish-coalescing': '...',
+   '@typescript-eslint/prefer-optional-chain': '...',
++  '@typescript-eslint/prefer-regexp-exec': '...',
+   '@typescript-eslint/prefer-string-starts-ends-with': '...',
+}
+```
+
+</details>
+
+<details>
+<summary>Code that generated those diffs</summary>
+
+```json title="package.json"
+{
+  "dependencies": {
+    "@typescript-eslint/eslint-plugin-7": "npm:@typescript-eslint/eslint-plugin@7",
+    "@typescript-eslint/eslint-plugin-8": "npm:@typescript-eslint/eslint-plugin@rc-v8"
+  },
+  "type": "module"
+}
+```
+
+```js
+import tseslint7 from '@typescript-eslint/eslint-plugin-7';
+import tseslint8 from '@typescript-eslint/eslint-plugin-8';
+
+function createDiffPatch(v7, v8) {
+  const v7Keys = new Set(Object.keys(v7));
+  const v8Keys = new Set(Object.keys(v8));
+  const output = ['{'];
+
+  for (const key of Array.from(new Set([...v7Keys, ...v8Keys])).sort((a, b) =>
+    trimSlash(a).localeCompare(trimSlash(b)),
+  )) {
+    const prefix = v7Keys.has(key) ? (v8Keys.has(key) ? ' ' : '-') : '+';
+
+    output.push(`${prefix}  '${key}': '...',`);
+  }
+
+  output.push('}');
+
+  return output.join('\n');
+}
+
+function trimSlash(text) {
+  return text.startsWith('@typescript-eslint/')
+    ? text.slice('@typescript-eslint/'.length)
+    : text;
+}
+
+const configNames = [
+  'recommended',
+  'recommended-type-checked',
+  'strict',
+  'strict-type-checked',
+  'stylistic',
+  'stylistic-type-checked',
+];
+
+for (const configName of configNames) {
+  console.log(configName);
+  console.log(
+    createDiffPatch(
+      tseslint7.configs[configName].rules,
+      tseslint8.configs[configName].rules,
+    ),
+  );
+}
+```
+
+</details>
```

Comment:
do we want to include this in the blog post? seems like a distraction.

Reply:
Heh, yeah, good point. It was more for me as the author than anybody else.